packages/wonder-blocks-tooltip/src/components/tooltip-popper.tsx

```diff
@@ -64,11 +78,42 @@ const filterPopperPlacement = (
     }
 };
 
+type SmallViewportModifier = Modifier<"smallViewport", object>;
+
+/**
+ * This is function calculates the the height of the popper
+ * vs. the height of the viewport. If the popper is larger
+ * than the viewport, it sets the popper isReferenceHidden
+ * state to false, to ensure the popper stays visible even if
+ * the reference is no longer in view. If the popper is less
+ * than the viewport, it leaves it as is so the popper will
+ * disappear if the reference is no longer in view.
+ */
+function _modifyPosition({state}: ModifierArguments<object>): void {
+    // Calculates the available space for the popper based on the placement
+    // relative to the viewport.
+    const popperHeight =
+        state.rects.popper.height + state.rects.reference.height;
+    const html = document.documentElement;
+    const minHeight = html.clientHeight;
+
+    if (minHeight < popperHeight && state.modifiersData.hide) {
+        state.modifiersData.hide = {
```

Comment:
This doesn't need to use `setState` or anything? I'm guessing not, I'm guessing it's Popper state and not React state. Just thought I'd double check.